### PR TITLE
CLI-1566 add support for C# on bitloops

### PR DESCRIPTION
**Pull request by threeguts**
## Summary
Added C# support

## Validation

- [X] I ran the relevant tests locally.
- [X] I included the executed commands and outcomes in the PR description.

## TDD RED Evidence Gate (when applicable)

Reference policy: `docs/tdd-red-evidence.md`

- [ ] This PR does **not** require RED evidence (explain why), or
- [ ] RED evidence exists on all required Jira test subtasks before implementation completion.
- [ ] RED evidence comments include command, failing result, and meaningful failure message.
- [ ] No tests were bypassed with `#[ignore]` or equivalent shortcuts.

## Jira

- [ ] Linked Jira issue(s) are updated with implementation notes and test results.
- [ ] Final status transitions match the task definition of done.

### CODE REVIEW

File: bitloops/src/adapters/languages/csharp/canonical.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+use crate::host::devql::CanonicalKindProjection;
+use crate::host::language_adapter::{CSharpKind, CanonicalMapping, LanguageKind, MappingCondition};
+
+pub(crate) static CSHARP_CANONICAL_MAPPINGS: &[CanonicalMapping] = &[
+    CanonicalMapping {
+        language_kind: LanguageKind::csharp(CSharpKind::Class),
+        projection: CanonicalKindProjection::Type,
```
Comment (giorgosnty) [inline bitloops/src/adapters/languages/csharp/canonical.rs:7]: I believe projection should be renamed to canonical_kind for consistency

File: bitloops/src/adapters/languages/csharp/canonical.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        condition: MappingCondition::Always,
```
Comment (giorgosnty) [inline bitloops/src/adapters/languages/csharp/canonical.rs:8]: condition should be removed unless useful for any reason

File: bitloops/src/adapters/languages/csharp/canonical.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        projection: CanonicalKindProjection::Enum,
+        condition: MappingCondition::Always,
+    },
+    CanonicalMapping {
+        language_kind: LanguageKind::csharp(CSharpKind::Method),
+        projection: CanonicalKindProjection::Method,
+        condition: MappingCondition::WhenInsideParent,
+    },
+    CanonicalMapping {
+        language_kind: LanguageKind::csharp(CSharpKind::Method),
```
Comment (giorgosnty) [inline bitloops/src/adapters/languages/csharp/canonical.rs]: Method is a method, let's remove this mapping

File: bitloops/src/adapters/languages/csharp/canonical.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        projection: CanonicalKindProjection::Type,
+        condition: MappingCondition::Always,
+    },
+    CanonicalMapping {
+        language_kind: LanguageKind::csharp(CSharpKind::Record),
+        projection: CanonicalKindProjection::Type,
+        condition: MappingCondition::Always,
+    },
+    CanonicalMapping {
+        language_kind: LanguageKind::csharp(CSharpKind::Delegate),
```
Comment (giorgosnty) [inline bitloops/src/adapters/languages/csharp/canonical.rs]: Delegate is something C# specific so let's remove it from canonical

File: bitloops/src/adapters/languages/csharp/edges.rs [hunk 5]
```diff
@@ -0,0 +1,464 @@
+use std::collections::{HashMap, HashSet};
+
+use anyhow::{Context, Result};
+use tree_sitter::Node;
+
+use super::extraction::{trimmed_node_text, using_target_name};
+use crate::host::devql::{
+    CallForm, CanonicalKindProjection, EdgeKind, ImportForm, RefKind, Resolution,
```
Comment (giorgosnty) [inline bitloops/src/adapters/languages/csharp/edges.rs:8]: Is implements missing?

File: bitloops/src/adapters/languages/csharp/test_support.rs [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+        let root = tree.root_node();
+        let bytes = source.as_bytes();
+
+        Ok(DiscoveredTestFile {
+            relative_path: relative_path.to_string(),
+            language: "csharp".to_string(),
+            reference_candidates: collect_csharp_import_paths(root, bytes)
+                .into_iter()
+                .map(ReferenceCandidate::SourcePath)
+                .collect(),
```
[the remark this answers is not in the document]
Comment (giorgosnty) [reply bitloops/src/adapters/languages/csharp/test_support.rs]: @threeguts Let's review this.

File: bitloops/src/host/language_adapter/kinds.rs [hunk 7]
```diff
@@ -308,13 +376,14 @@ impl TryFrom<&str> for LanguageKind {
...
     fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let csharp = CSharpKind::from_tree_sitter_kind(value).map(Self::CSharp);
         let rust = RustKind::from_tree_sitter_kind(value).map(Self::Rust);
         let ts_js = TsJsKind::from_tree_sitter_kind(value).map(Self::TsJs);
         let python = PythonKind::from_tree_sitter_kind(value).map(Self::Python);
         let go = GoKind::from_tree_sitter_kind(value).map(Self::Go);
         let java = JavaKind::from_tree_sitter_kind(value).map(Self::Java);
 
-        match [rust, ts_js, python, go, java]
+        match [rust, ts_js, python, go, java, csharp]
```
[the remark this answers is not in the document]
Comment (giorgosnty) [reply bitloops/src/host/language_adapter/kinds.rs:386]: @threeguts do you understand this comment? I don't but is it something that makes sense?

File: bitloops/src/adapters/languages/csharp/extraction.rs [hunk 8]
```diff
@@ -0,0 +1,488 @@
...
+                            seen,
+                            node,
+                            content,
+                            CSharpArtefactDescriptor {
+                                language_kind: LanguageKind::csharp(CSharpKind::Field),
+                                name,
+                                symbol_fqn,
+                                parent_symbol_fqn: parent_fqn.map(str::to_string),
+                            },
+                        );
```
[the remark this answers is not in the document]
Comment (giorgosnty) [reply bitloops/src/adapters/languages/csharp/extraction.rs:233]: @threeguts this seems valid and we can add a specific test for that where we have a variable declarator with a,b and an edge referencing only b for example not a/

File: bitloops/src/adapters/languages/csharp/test_support.rs [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+            suites: collect_csharp_test_suites(root, bytes, relative_path),
```
[the remark this answers is not in the document]
Comment (giorgosnty) [reply bitloops/src/adapters/languages/csharp/test_support.rs:74]: @threeguts let's see what this is about too and assess it

File: bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs [hunk 10]
```diff
@@ -759,3 +759,188 @@ class Greeter extends Base implements Runner {
...
+            true,
+            Some("interface"),
+        ),
+        (
+            LanguageKind::csharp(CSharpKind::Enum),
+            false,
+            true,
+            Some("enum"),
+        ),
+        (LanguageKind::csharp(CSharpKind::Method), false, true, None),
```
Comment (giorgosnty) [inline bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs]: This means method not inside parent(class?) and in this case we don't map it to something? What is the case? I guess every method is by default inside a class, right?

File: bitloops/src/host/devql/tests/mapping_tests/registry_and_resolution.rs [hunk 11]
```diff
@@ -69,8 +79,81 @@ fn devql_language_adapter_registry_resolves_built_in_pack_implementations() {
...
+    assert!(
+        csharp_artefacts
+            .iter()
+            .any(|artefact| artefact.name == "UserService"),
+        "csharp built-in registry pack should surface type artefacts"
+    );
+    let csharp_edges = csharp_pack
+        .extract_dependency_edges(csharp_content, "src/UserService.cs", &csharp_artefacts)
+        .expect("extract csharp dependency edges via language adapter registry");
+    assert!(
```
Comment (giorgosnty) [inline bitloops/src/host/devql/tests/mapping_tests/registry_and_resolution.rs]: Let's assert specific numbers(1 calls, 2 imports etc) and the from to in the edges so that it's clear the test pass, otherwise it could pass accidentally.

File: bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs (hunk 10 above)
Comment (threeguts) [reply bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs]: Yes methods are by default inside a class, this prevents misclassifying orphan methods from being canonically treated the same as properly-owned methods.

Comment (giorgosnty) [reply bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs]: But can we have orphan methods is the question? if not let's remove this case.

Comment (threeguts) [reply bitloops/src/host/devql/tests/mapping_tests/canonical_mappings.rs]: ok i removed it since it was redundant

File: bitloops/src/host/language_adapter/kinds.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (threeguts) [reply bitloops/src/host/language_adapter/kinds.rs:386]: the c# branch it asks for exists

File: bitloops/src/adapters/languages/csharp/test_support.rs (hunk 9 above)
[the remark this answers is not in the document]
Comment (threeguts) [reply bitloops/src/adapters/languages/csharp/test_support.rs:74]: Already fixed

**Review by giorgosnty — APPROVED** [review]
